### PR TITLE
QGIS logo for composer

### PR DESCRIPTION
**Pull request by timlinux**
## Description

There has been some community work to develop nice 'made with QGIS' badges that can be added to composer layouts to promote the fact that a map was developed using QGIS. It is, unfortunately, mostly invisible to users. Then I also notced that the QGIS logo itself isn't easy to find. So this feature tries to fix that. 

It adds a new group under 'SVG' in the composer picture item properties:

image

Currently these logos are included:

1. The QGIS icon (without text) in black with parameterized options for fill, outline, outline width and transparency.
1. The QGIS logo (with text) in colour
1. The OSGeo logo in colour
1. A 'made with QGIS' monochrome logo created by QGIS community contributor @klakar (and included in this feature with his permission). This logo is parameterized so the outline and fill colours can be changed by the users. Default colour is black with a 0 point outline.
1. A 'made with QGIS' colour logo created by QGIS community contributor @klakar (and included in this feature with his permission). This one is not parameterized.

The following animation demonstrates the workflow of using the new resources. Note the animation shows some dithering due to the number of colours, the actual logos look much nicer in real use).

qgis-logo-patch

### CODE REVIEW

**Comment by nirvn** [comment]
@timlinux , I like the idea. Seeing the folder QGIS in all-caps sitting next to lowercase folder names tickles me. One suggestion: what about renaming the folder to 'logos', and have QGIS logos alongside osgeo?

**Comment by andreasneumann** [comment]
> @timlinux , I like the idea. Seeing the folder QGIS in all-caps sitting next to lowercase folder names tickles me. One suggestion: what about renaming the folder to 'logos', and have QGIS logos alongside osgeo?

I like this idea to also have an OSGEO logo - and a "logos" folder. There could be more in the future. Perhaps someone wants to credit "proj", because of some special projection being used ...

**Comment by timlinux** [comment]
ok good suggestions @nirvn and @andreasneumann, will do.

**Comment by timlinux** [comment]
qgis-logo-patch

Updates from last commit illustrated above

**Comment by roya0045** [comment]
Is it worth adding logos of other dependencies (GRASS,SAGA, GDAL, MDAL,PDAL,etc)?

**Comment by timlinux** [comment]
> Is it worth adding logos of other dependencies (GRASS,SAGA, GDAL, MDAL,PDAL,etc)?

Happy to do that if others think it is useful, but for me I am struggling to think of a situations where I would put those logos onto a map I create.....

**Comment by roya0045** [comment]
@timlinux As with other logos, mostly as a showcase. I just wanted to raise the proposition as @andreasneumann mentioned adding related logos.

**Comment by nyalldawson** [comment]
> The QGIS logo (without text) in black, white and QGIS green

Can't we parameterize this one, and set the default fill to QGIS green? It'd be more flexible + avoid the duplicate copies

**Comment by nyalldawson** [comment]
It looks from the screencast like the svg viewboxes for the logos aren't correct -- I think that's why the logo is clipped in the list view. Can you fix this?

File: images/svg/logos/qgis-logo-made-with-monochrome.svg [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+     fill="param(fill) #000" stroke="param(outline) #000" stroke-width="param(outline-width) 0"
```
Comment (nyalldawson) [inline images/svg/logos/qgis-logo-made-with-monochrome.svg:103]: this is missing the fill-opacity and outline-opacity parameters too

**Comment by timlinux** [comment]
Thanks for the feedback @nyalldawson 

I have made your proposed changes.

qgis-logo-patch

Only the viewbox still escapes me - it clips the QGIS Icon. I tried googling around and poking around in Inkscape to fix it without success.

**Comment by andreasneumann** [comment]
@timlinux - for the OSGEO logo replace the second line with this:

```
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 560 180" width="100%" height="100%">
```

If you set width and height to 100% then it should scale well. The viewBox acts like a paper drawing space. All of the coordinates used in the geometries need to fit into the viewBox if you want to show all of the graphics.

You can also have negative coordinates in the viewBox - that is totally possible.

**Comment by klakar** [comment]
Fiddled around with the code...

```
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<svg
   xmlns:dc="http://purl.org/dc/elements/1.1/"
   xmlns:cc="http://creativecommons.org/ns#"
   xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
   xmlns:svg="http://www.w3.org/2000/svg"
   xmlns="http://www.w3.org/2000/svg"
   version="1.1"
   id="Layer_1"
   x="0px"
   y="0px"
   width="127.86523"
   height="124.43359"
   viewBox="0 0 127.86523 124.43359"
   enable-background="new 0 0 128 128"
   xml:space="preserve"><metadata
   id="metadata976"><rdf:RDF><cc:Work
       rdf:about=""><dc:format>image/svg+xml</dc:format><dc:type
         rdf:resource="http://purl.org/dc/dcmitype/StillImage" /><dc:title></dc:title></cc:Work></rdf:RDF></metadata><defs
   id="defs974" />
<path
   fill="param(fill) #589632" opacity="param(fill-opacity)" stroke="param(outline) #589632" stroke-with="param(outline-width) 0" stroke-opacity="param(outline-opacity)"
   id="polygon965"
   d="M 63.849609,2.9403944e-8 C 27.363609,-9.999706e-4 0,25.504891 0,59.962891 c 0,34.625 27.362656,60.638669 63.847656,60.638669 9.417,0 16.069969,-1.46972 23.042969,-3.76172 L 68.923828,99.361324 c -1.165,0.242 -1.769547,0.15625 -4.685547,0.15625 -20.866,0 -38.611328,-17.158246 -38.611328,-39.406246 0,-22.248 17.550328,-39.027343 38.611328,-39.027343 21.061,0 37.832029,16.780343 37.832029,39.027343 0,3.619 -0.4502,7.099438 -1.2832,10.398438 l 19.3125,19.310547 c 4.979,-8.726 7.76562,-18.869422 7.76562,-29.857422 C 127.86523,25.673891 100.50261,1.2940394e-7 63.849609,2.9403944e-8 Z M 52.613281,52.433594 v 18.121094 l 16,15.802734 V 67.433594 h 18.277344 l -15.193359,-15 z m 24,23 v 18.408203 l 30.529299,30.591793 h 19.4707 V 106.86523 L 94.488281,75.433594 Z" />
   
</svg>
```

**Comment by andreasneumann** [comment]
Here is a better version of the file qgis-icon.svg:

Note that by adding extra space on the left and top you avoid "cutoffs" if the stroke is made is thicker - hence the negative values in the viewBox

Adobe Illustrator and - to some extent also Inkscape - add a lot of stupid stuff to the files that are useless, such as the DOCTYPE, metadata/RDF without content, made with Illustrator useless comments and such --> best to remove this as well.

```
<?xml version="1.0" encoding="utf-8"?>
<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="100%" height="100%" viewBox="-5 -5 135 135" xml:space="preserve">
<polygon 
fill="param(fill) #589632" fill-opacity="param(fill-opacity)" stroke="param(outline) #589632" stroke-opacity="param(outline-opacity)" stroke-width="param(outline-width) 0"
points="68.613,69.625 86.891,69.625 71.697,54.625 52.613,54.625 52.613,72.746 68.613,88.548 "/>
<polygon 
fill="param(fill) #589632" fill-opacity="param(fill-opacity)" stroke="param(outline) #589632" stroke-opacity="param(outline-opacity)" stroke-width="param(outline-width) 0"
points="126.613,109.057 94.488,77.625 76.613,77.625 76.613,96.033 107.143,126.625 126.613,126.625 "/>
<path 
fill="param(fill) #589632" fill-opacity="param(fill-opacity)" stroke="param(outline) #589632" stroke-opacity="param(outline-opacity)" stroke-width="param(outline-width) 0"
d="M68.923,101.552c-1.165,0.242-1.769,0.157-4.685,0.157c-20.866,0-38.612-17.158-38.612-39.406
	c0-22.248,17.551-39.027,38.612-39.027s37.833,16.78,37.833,39.027c0,3.619-0.451,7.099-1.284,10.398L120.1,92.012
	c4.979-8.726,7.765-18.869,7.765-29.857c0-34.289-27.363-59.963-64.016-59.963C27.363,2.191,0,27.696,0,62.154
	c0,34.625,27.363,60.638,63.848,60.638c9.417,0,16.069-1.469,23.042-3.761L68.923,101.552z"/>
</svg>
```

**Comment by timlinux** [comment]
Thanks @andreasneumann and @klakar . I incorporated Andreas's suggestions (which I guess override yours @klakar). I am still not seeing any difference in the list view layout for the icons.

 
image

**Comment by timlinux** [comment]
@nyalldawson Can we go ahead and merge this?

**Comment by agiudiceandrea** [comment]
> Only the viewbox still escapes me - it clips the QGIS Icon

> I am still not seeing any difference in the list view layout for the icons.

Hi @timlinux, this issue usually happens when SVGs with square and rectangular aspect ratios are in the same folder.
Just making them all square or all rectangular solves the issue:
mixed
image 
all square
image
all rectangular
image

**Comment by nyalldawson** [comment]
@agiudiceandrea it's fixed here: https://github.com/qgis/QGIS/pull/40577

**Pull request merged**